pkg/build/webhook/generic/generic.go

```diff
@@ -43,67 +43,74 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 	}
 
 	if buildCfg.Spec.Source.Git == nil {
-		glog.V(4).Infof("No git source defined for BuildConfig %s/%s, but triggering anyway", buildCfg.Namespace, buildCfg.Name)
-		return revision, envvars, true, err
+		warning := webhook.NewWarning("BuildConfig has no git source defined, ignoring and continuing with build")
```

Comment:
i don't think we need this warning anymore.  once upon a time git source was pretty much it, but now you could have pipeline builds, or builds that take input from images and builds that embed docker files.  if we're going to keep it, it needs to check for all the possible sources of "source" input, basically to ensure it's not a pure binary-only buildconfig.